lib/cache_shoe/wrapper.rb

```diff
@@ -23,15 +24,13 @@ def module_instance
 
     def wrap_the_method_to_cache
       cached_method = method_name
-
+      model_class = model
       module_instance.send :define_method, cached_method do |*args, &block|
         scope = Scope.new(
-          object: self,
-          cached_method: cached_method,
+          model_class: model_class || self.class,
```

Comment:
I think if we need to access `self.class` while initializing `Scope`, we should just pass `self` in and let it decide.